DataFormats/ForwardDetId/interface/HGCalDetId.h

```diff
@@ -59,6 +59,27 @@ class HGCalDetId : public DetId {
   static const HGCalDetId Undefined;
 };
 
+namespace HGCalDetIdUtil
+{
+  // Functions to work on raw HGCAlDetIds
+  int subdetIdOf(uint32_t id);
+  int cellOf(uint32_t id);
+  int waferOf(uint32_t id);
+  int waferTypeOf(uint32_t id);
+  int layerOf(uint32_t id);
+  int zsideOf(uint32_t id);
+
+  void setCellOf(uint32_t& id, int cell);
+  void setWaferOf(uint32_t& id, int mod);
```

Comment:
@jbsauvan - looking again at these changes and how you use the functions defined - I don't see why you don't just use HGCalDetId instead of the uint32_t in your functions below. There should be no overhead in doing that (and it seems all the functionality needed is already there.